packages/functions/transform/src/merge-request-metrics.ts

```diff
@@ -1122,9 +1121,14 @@ export async function run(extractMergeRequestId: number, ctx: RunContext) {
   });
 
   const timezone = await getTimezoneInMinutes(ctx.tenantDatabase);
-  let timezoneInMilliseconds = 0;
+  const currDate = new Date();
+  const options: Intl.DateTimeFormatOptions = { timeZone: timezone };
+  const hqTime = new Date(currDate.toLocaleString('en-US', options));
+  const timezoneOffset = (currDate.getTime() - hqTime.getTime()) / (1000 * 60);
+  const timezoneInMinutes = Math.round(timezoneOffset);
 
-  timezoneInMilliseconds = timezone * 60 * 1000;
+  let timezoneInMilliseconds = 0;
+  timezoneInMilliseconds = timezoneInMinutes * 60 * 1000;
   
   const timestampsWithTimezone = new Set(
     Array.from(mergeRequestTimestamps).map(timestamp => timestamp + timezoneInMilliseconds)
```

Comment:
timezoneInMilliSeconds depends on the date being transformed

Reply:
if you want to stay with this approach, I'd prefer to move the code a different file that exports the function i.e. `getLocaleTimezoneOffset` and write a few tests for it so we are sure it works